phac-nml/irida-next
# Add data export model

## What does this PR do and why?
This PR adds the new `DataExports` model which will be used to track user exports.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
1. In the rails console `rails c`, run `bin/rails db:migrate` to migrate the new `DataExports` table
2. Assign a user to the variable `user`
3. Set a datetime `date = Time.now + 3.days`
4. Test adding a new `DataExport` in the console with the command `DataExport.new(user: user, name: 'test name', export_type: 'sample', status: 'processing', expires_at: date, email_notification: true, export_parameters: {ids: [1,2,3]}).save`
5. Check the export exists under `user.data_exports`
6. Test that you cannot add a new `DataExport` if the following fields are empty/missing: `user`, `export_type`, `status`.
7. Test that you can add a new `DataExport` if the following fields are empty/missing: `name`, `email_notification`, `export_parameters`, `expires_at`.
8. Ensure tests run locally

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
